echopype/echodata/echodata.py

```diff
@@ -268,43 +268,42 @@ def __setattr__(self, __name: str, __value: Any) -> None:
     def update_platform(
         self,
         extra_platform_data: xr.Dataset,
-        time_dim="time",
+        variable_mappings=Dict[str, str],
         extra_platform_data_file_name=None,
     ):
         """
         Updates the `EchoData["Platform"]` group with additional external platform data.
 
-        `extra_platform_data` must be an xarray Dataset.
-        The name of the time dimension in `extra_platform_data` is specified by the
-        `time_dim` parameter.
-        Data is extracted from `extra_platform_data` by variable name; only the data
-        in `extra_platform_data` with the following variable names will be used:
-            - `"pitch"`
-            - `"roll"`
-            - `"vertical_offset"`
-            - `"latitude"`
-            - `"longitude"`
-            - `"water_level"`
-        The data inserted into the Platform group will be indexed by a dimension named
-        `"time1"`.
+        `extra_platform_data` must be an xarray Dataset. Data is extracted from
+        `extra_platform_data` by variable name. Only data assigned to a pre-existing
+        (but possibly all-nan) variable name in Platform will be processed. These
+        Platform variables include latitude, longitude, pitch, roll, vertical_offset, etc.
+        Different external variables may be dependent on different time dimensions, but
+        latitude and longitude (if specified) must share the same time dimension. New
```

Comment:
I don't see a check for this in the code. How about adding a simple defensive check to make sure this is true? Seems like once you have `mappings_expanded` this would be straightforward to do.

Reply:
Right! I'll add that. I thought I had done it, but I guess not.

Reply:
Done, in my latest commits.